### ISSUE TITLE
Kitty sometimes exit with "line height too large after adjustment" on Wayland WM Sway

### ISSUE THREAD

**Issue by ya-isakov** [kovidgoyal/kitty#1380]
Hello! I've found that sometimes, when I unplug my external monitor (or plug it and disable internal display of laptop), kitty closes. I'm using Sway WM from master, kitty is Wayland-native (started with KITTY_ENABLE_WAYLAND=1 kitty). The error is `line height too large after adjustment`

**Comment by kovidgoyal** [comment]
That will happen when the calculated line height becomes more than 1000 pixels. This in turn will happen if the OS reports incorrect DPI values to kitty. See the checkScaleChange() function in glfw/wl_window.c you will likely find some invalid value of scale there.

**Comment by kovidgoyal** [comment]
no followup

**Comment by ya-isakov** [comment]
Sorry for not answering. I've tried to trace this problem, and I'm not quite sure if it's WM or not. It looks like in some rare cases, wl.monitors[0] is broken, so it's scale could be ~4000 or ~2000000, and name is also something unexpected. I could easily reproduce this bug, and I've tried to trace every possible place where wl.monitors is changing, but I failed. If you have any tip/suggestion for further investigation, I would be happy to continue it.
